### ISSUE TITLE
Enhancement for nlu.yml/python intent object 

### ISSUE THREAD

**Issue by joggerjoel** [RasaHQ/rasa#8557]
**Description of Problem**:
- intent: chitchat/thanks
  examples: |
    - thx
    - Awesome, thanks!
    - thanks mate
    - thank you
In nlu you can have intent/subintent; but the python tracker object doesn’t pick it up I only can see tracker.latest_message[‘intent’][‘name’]) which doesn’t provide enough details to make a choice if I use intent/subintent. Please think about adding this exposure.

Thanks.

**Overview of the Solution**:
tracker.latest_message[‘intent’][‘subname’]

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @samsucik will get back to you about it soon! ✨

**Comment by samsucik** [comment]
Hey @joggerjoel, is the aim to be able to condition on subintents (response keys) when choosing next action?

**Comment by joggerjoel** [comment]
Yes. I’d like more granular logic to respond/react especially when
confidence is low.

**Comment by samsucik** [comment]
I see. This is not currently possible, and it would require a fair bit of change in the codebase... Retrieval intents were designed to give more granular/context-specific responses while keeping the training stories simple. That's why Rasa Core ignores the response key and predicts the next action based on the base intent only. So, retrieval intents don't enable true intent hierarchy. However, you're definitely the first one interested in this (just search for "hierarchical intents" here or in the Rasa forum) -- perhaps we (or a contributor) will make it part of Rasa in the future?

**Comment by samsucik** [comment]
By the way, which confidence do you mean when you say "especially when
confidence is low"? What's the main problem you're trying to solve?
